src/com/sap/piper/tools/ToolType.groovy

```diff
@@ -0,0 +1,38 @@
+package com.sap.piper.tools
+
+
+enum ToolType {
+
+    JAVA ('Java', ToolKey.JAVA, ToolExecutable.JAVA, ToolVersion.JAVA), 
+    MTA ('SAP Multitarget Application Archive Builder', ToolKey.MTA, ToolExecutable.MTA, ToolVersion.MTA),
+    NEO ('SAP Cloud Platform Console Client', ToolKey.NEO, ToolExecutable.NEO, ToolVersion.NEO),
+    CM_CLI ('Change Management Command Line Interface', ToolKey.CM_CLI, ToolExecutable.CM_CLI, ToolVersion.CM_CLI)
+
+    final def name
+    final def key
+    final def executable
+    final def version
```

Comment:
I would use types everywhere, but this was discussed and the decision was to not use types.

Reply:
We introduce enums. Enums are cool since you know the values you will encounter: either one of the enum values or null. Now we have an enum, but we have def, still accepting a strings or whatever. When having an enum it is reasonable to check early if we have an enum. Otherwise the code will continue and failing at a later point, which is always harder than required to troubleshoot. IMO it makes sense to use type in conjunction with enums. But if you think that violates against team rules, we can keep it like this. It is only a suggestion, this it not a reason for not merging the change.